jujugui/static/gui/src/app/components/basic-table/basic-table.js

```diff
@@ -198,10 +208,15 @@ BasicTable.propTypes = {
     }).isRequired).isRequired,
     // Content to be displayed when the row is toggled.
     expandedContent: PropTypes.any,
+    // Set the expanded content state from outside the table.
+    expandedContentExpanded: PropTypes.bool,
     // Extra data that can be used when ordering, sorting etc.
     extraData: PropTypes.any,
     // The row key, used for React indexing and sorting.
-    key: PropTypes.string.isRequired
+    key: PropTypes.string.isRequired,
+    // Whether the row can be toggled by clicking on it. If this is not set the
+    // value will be controlled by whether there is expandable content.
```

Comment:
Why would you have expandable content but not want it to be toggled?

Reply:
This is if you want to have it controlled by an external button. In our case we want to click on the Edit item in the context menu rather than clicking on the row.